src/bin/cql-stress-cassandra-stress/settings/param/distribution.rs

```diff
@@ -0,0 +1,153 @@
+use std::collections::HashMap;
+
+use anyhow::Result;
+
+use crate::java_generate::distribution::DistributionFactory;
+
+use super::Param;
+
+/// Distribution-specific arguments parser.
+trait DistributionParser: Sync {
+    fn parse(&self, distribution_args: &[&str]) -> Result<Box<dyn DistributionFactory>>;
+    fn specification_and_description(&self) -> (&'static str, &'static str);
+}
+
+struct DistributionParam {
+    value: Option<Box<dyn DistributionFactory>>,
+}
+
+lazy_static! {
+    static ref DIST_PARSERS: HashMap<&'static str, Box<dyn DistributionParser>> = {
+        let map = HashMap::new();
+        map
+    };
+}
```

Comment:
I think this is overengineered. Why not have a function that does pattern matching one the string and returns a distribution parser?

Reply:
Yeah, I can see your point. I only did it for the "efficiency" purposes so we don't have to return a `Box`'ed `DistributionParser` with each invocation of such function. I wanted to allocate each parser once.

But as you said, this might be overengineered, since there wouldn't be many allocations of `DistributionParser` during parsing anyway.

Reply:
AFAIK `Box<T>` where `T` is a zero-sized type does not allocate.